src/signing.rs

```diff
@@ -495,6 +505,18 @@ where
     }
 }
 
+/// Equivalent to [`SigningKey::sign_prehashed`] with `context` set to [`Some`]
+/// containing [`Context::value`].
+#[cfg(feature = "digest")]
+impl<D> DigestSigner<D, Signature> for Context<'_, '_, SigningKey>
+where
+    D: Digest<OutputSize = U64>,
```

Comment:
I mean, I've never explicitly wanted to go out of the way to support non SHA-512 hashes. It can potentially be hardcoded to just `sha2::Sha512`.

Allowing a generic parameter does allow plugging in alternative implementations of SHA-512 though, such as a wrapper for a cryptographic accelerator.

We've tried to constrain the function used in other repos using `AssociatedOid` but unfortunately const generics aren't powerful enough to use the type system to enforce that, e.g. `where D: AssociatedOid<OID = {{ SHA512_OID }}>`, at least on stable.

Reply:
Ohh ok. That should definitely be documented somewhere. Maybe it is and I missed it. I can make the change later today